src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -401,16 +416,21 @@ public async Task<uint256> Rewind()
                 if (this.blockHash == this.innerBlockHash)
```

Comment:
This whole segment I don't understand. Could you please explain the logic here? For example why this if stayed in the code and the other if was removed?

Reply:
That shouldn't be here, good spotting. I forgot to set `blockHash`. Changed now